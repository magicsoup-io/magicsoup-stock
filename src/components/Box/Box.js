import styled from 'styled-components'
import {
  space,
  color,
  width,
  flex,
  order,
  alignSelf,
  fontSize,
} from 'styled-system'

const themed = key => props => props.theme[key]

const Box = styled('div')`
  box-sizing: border-box;
  ${space}
  ${width}
  ${fontSize}
  ${color}
  ${flex}
  ${order}
  ${alignSelf}
  ${themed('Box')}
`
/** @component */
export default Box

Box.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  ...fontSize.propTypes,
  ...color.propTypes,
  ...flex.propTypes,
  ...order.propTypes,
  ...alignSelf.propTypes,
}
