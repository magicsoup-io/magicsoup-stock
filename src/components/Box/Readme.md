Basic Box:

```jsx
<Box
  p={5}
  fontSize={6}
  width={[ 1, 1, 1/2 ]}
  color="white"
  bg="primary">
  Box
</Box>
```
