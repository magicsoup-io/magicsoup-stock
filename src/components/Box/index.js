export { default } from './Box'

// import Box from './Box'

// export default Box
