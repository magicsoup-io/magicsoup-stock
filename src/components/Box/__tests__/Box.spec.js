import React from 'react'
import renderer from 'react-test-renderer'

import Box from '../Box'

it('renders <Box /> without throwing', () => {
  const tree = renderer.create(<Box> I am a Box</Box>).toJSON()
  expect(tree).toMatchSnapshot()
})
