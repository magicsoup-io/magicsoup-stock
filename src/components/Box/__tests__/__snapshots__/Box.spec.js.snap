// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Box /> without throwing 1`] = `
<div
  className="sc-bdVaJa iobDuz"
>
   I am a Box
</div>
`;
